src/doc/src/reference/profiles.md

```diff
@@ -293,7 +294,7 @@ The default settings for the `release` profile are:
 opt-level = 3
 debug = false
 split-debuginfo = '...'  # Platform-specific.
-strip = "none"
+strip = "debuginfo" 
```

Comment:
This seems incorrect to me. The default value of `strip` for release profile is still `none`.

Reply:
Is `strip` by default will be equal to `none` for the release profile if `debug` is set to `false`? I read #13677, #13257, #13763, and #13638 but still didn't get it. Thanks in advance!

Reply:
The behavior of auto-stripping debuginfo is kinda dynamic. The default release profile settings is still static and `strip` is none when you want to inherit it.

Share these pieces of code for clarity.

https://github.com/rust-lang/cargo/blob/94aa7fb1321545bbe922a87cb11f5f4559e3be63/src/cargo/core/profiles.rs#L719-L728

https://github.com/rust-lang/cargo/blob/94aa7fb1321545bbe922a87cb11f5f4559e3be63/src/cargo/core/profiles.rs#L630-L651

Reply:
> The default release profile settings is still static and strip is none when you want to inherit it.

Hmm.
According to [Rust Blog](https://blog.rust-lang.org/2024/03/21/Rust-1.77.0.html#enable-strip-in-release-profiles-by-default):
> Cargo [profiles](https://doc.rust-lang.org/stable/cargo/reference/profiles.html) which do not enable [debuginfo](https://doc.rust-lang.org/stable/cargo/reference/profiles.html#debug) in outputs (e.g., debug = 0) will enable strip = "debuginfo" by default.

And according to [Cargo Book](https://doc.rust-lang.org/cargo/reference/profiles.html#release):
> [profile.release]
...
debug = false

I made a small test on 1.78.0 and release profile seems indeed have set `strip = "debuginfo"` by default.
Not sure that behaviour should be, but it seems for me also that documentation is not updated and #13677 make sense.
Or am I missing something here?
Thanks!

Reply:
See also <https://github.com/rust-lang/cargo/pull/13638#issuecomment-2018594383> and <https://github.com/rust-lang/cargo/pull/13677#discussion_r1545924873>.

Reply:
The settings are static. The actual flags passed to rustc are deferred later when Cargo see if it can safely strip std or not.

Reply:
> The default is still strip=none if you turn on debug for release profile. strip=debuginfo only happens for release profile when debug is off.

But `debug` is off by default for the release profile, so `strip="debuginfo"` will take place, no? 🤔
Not sure I get it correctly, but maybe it would be great to add more context to `The Cargo Book`.
Thanks for the explanation!